packages/nodes-base/nodes/Totp/Totp.node.ts

```diff
@@ -0,0 +1,169 @@
+
+export class Totp implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'TOTP',
+		name: 'totp',
+		icon: 'fa:fingerprint',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{ $parameter["operation"] }}',
+		description: 'Generate a time-based one-time password',
+		defaults: {
+			name: 'TOTP',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'totpApi',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				noDataExpression: true,
+				options: [
+					{
+						name: 'Generate Secret',
+						value: 'generateSecret',
+						action: 'Generate secret',
+					},
+				],
+				default: 'generateSecret',
+			},
+			{
+				displayName: 'Additional Options',
+				name: 'additionalOptions',
+				type: 'collection',
+				displayOptions: {
+					show: {
+						operation: ['generateSecret'],
+					},
+				},
+				default: {},
+				placeholder: 'Add Option',
+				options: [
+					{
+						displayName: 'Algorithm',
+						name: 'algorithm',
+						type: 'options',
+						default: 'SHA1',
+						description: 'HMAC hashing algorithm. Defaults to SHA1.',
+						options: [
+							{
+								name: 'SHA1',
+								value: 'SHA1',
+							},
+							{
+								name: 'SHA224',
+								value: 'SHA224',
+							},
+							{
+								name: 'SHA256',
+								value: 'SHA256',
+							},
+							{
+								name: 'SHA3-224',
+								value: 'SHA3-224',
+							},
+							{
+								name: 'SHA3-256',
+								value: 'SHA3-256',
+							},
+							{
+								name: 'SHA3-384',
+								value: 'SHA3-384',
+							},
+							{
+								name: 'SHA3-512',
+								value: 'SHA3-512',
+							},
+							{
+								name: 'SHA384',
+								value: 'SHA384',
+							},
+							{
+								name: 'SHA512',
+								value: 'SHA512',
+							},
+						],
+					},
+					{
+						displayName: 'Digits',
+						name: 'digits',
+						type: 'number',
+						default: 6,
+						description: 'Number of digits in the generated TOTP code. Defaults to 6 digits.',
+					},
+					{
+						displayName: 'Period',
+						name: 'period',
+						type: 'number',
+						default: 30,
+						description:
+							'How many seconds the generated TOTP code is valid for. Defaults to 30 seconds.',
+					},
+				],
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: INodeExecutionData[] = [];
+
+		const operation = this.getNodeParameter('operation', 0);
+		const credentials = (await this.getCredentials('totpApi')) as { label: string; secret: string };
+
+		if (!credentials.label.includes(':')) {
+			throw new NodeOperationError(this.getNode(), 'Malformed label - expected `issuer:username`');
+		}
+
+		const additionalOptions = this.getNodeParameter('additionalOptions', 0) as {
+			algorithm: string;
+			digits: number;
+			period: number;
+		};
```

Comment:
Is it make sense to do this in loop for each incoming item?

Reply:
Decided against this because token generation depends also on cred data, which is fixed.